lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/dataset/ReferenceDatasetUtils.scala

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.lightgbm.dataset
+
+import com.microsoft.azure.synapse.ml.lightgbm.swig.SwigUtils
+import com.microsoft.azure.synapse.ml.lightgbm._
+import com.microsoft.ml.lightgbm._
+import org.apache.spark.sql._
+import org.slf4j.Logger
+
+
+object ReferenceDatasetUtils {
+  def createReferenceDatasetFromSample(datasetParams: String,
+                                       featuresCol: String,
+                                       numRows: Long,
+                                       numCols: Int,
+                                       sampledRowData: Array[Row],
+                                       measures: InstrumentationMeasures,
+                                       log: Logger): Array[Byte] = {
+    log.info(s"For reference Dataset creation, received sample data with ${sampledRowData.length} samples")
+
+    // Pre-create allocated native pointers so it's easy to clean them up
+    val datasetVoidPtr = lightgbmlib.voidpp_handle()
+    val lenPtr = lightgbmlib.new_intp()
+    val bufferHandlePtr = lightgbmlib.voidpp_handle()
+
+    val sampledData: SampledData = new SampledData(sampledRowData.length, numCols)
```

Comment:
nit: remove the type spec

Reply:
nit: Could `SampledData` be a case class with `apply`?

Reply:
Can you elaborate on the second comment?  SampledData has a lot of methods. I'm not sure what you mean by make it a case class. I am relatively new to Scala, but for me "case class" is just for a simple POCO class.

Reply:
Just change `class` to `case class` and remove the above `new`. It should work.